Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m

```diff
@@ -39,19 +39,37 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
         }
         return;
     }
-    
+
     BOOL hasUpdated = NO;
-    for (NSString *key in response.data) {
-        NSInteger credits = [response.data[key] integerValue];
-        
-        BNCPreferenceHelper *preferenceHelper = [BNCPreferenceHelper preferenceHelper];
-        if (credits != [preferenceHelper getCreditCountForBucket:key]) {
+    BNCPreferenceHelper *preferenceHelper = [BNCPreferenceHelper preferenceHelper];
+    NSDictionary *currentCreditDictionary = [preferenceHelper getCreditDictionary];
+    NSArray *responseKeys = [response.data allKeys];
+    NSArray *storedKeys = [currentCreditDictionary allKeys];
+
+    if ([responseKeys count]) {
+        for (NSString *key in response.data) {
+             NSInteger credits = [response.data[key] integerValue];
+
+             BNCPreferenceHelper *preferenceHelper = [BNCPreferenceHelper preferenceHelper];
+             if (credits != [preferenceHelper getCreditCountForBucket:key]) {
+                 hasUpdated = YES;
+             }
+
+             [preferenceHelper setCreditCount:credits forBucket:key];
+        }
+        for(NSString *key in storedKeys) {
+            if(![response.data objectForKey:key]) {
+                [preferenceHelper removeCreditCountForBucket:key];
+                hasUpdated = YES;
+            }
+        }
+    } else {
+        if ([storedKeys count]) {
```

Comment:
It's strictly for the case that user A had "default" : x, and suddenly doesn't have default anymore - usually a redeem takes care of this, but if a user didn't redeem inside the app, this could pop up